add end-end-tests

kaushik-rohit opened the pull request:
## What? Why?
Fix # .Add more units tests

Changes proposed in this pull request:
- a add more tests and output data related to it
- b fix tavis.yml to run cd and test in a single command

## Checks
- [x] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Any images?

## Notify reviewers
@rohangoel96

rohangoel96 commented on test/tests.py:
```diff
@@ -1,29 +1,161 @@
+dir = os.path.dirname(__file__)
 
+log_directory = os.path.join(dir, 'data/input/')
+output_directory = os.path.join(dir, 'data/output/')
```
this is not the output directory ( you aren't putting the outputting anything here). expected_output_directory makes sense

rohangoel96 commented on test/tests.py:
```diff
@@ -1,29 +1,161 @@
 nicks, nick_same_list = nickTracker.nick_tracker(log_data)
 
+def compare_graph_outputs (generated_output, stored_output_file_name):
+
```
don't add extra enters. It makes very difficult to read the code.
and use 2 enters between function definitions. Aren't you using a linter?

rohangoel96 commented on test/tests.py:
```diff
@@ -1,29 +1,161 @@
+    return False
+
+def csv_to_list (csv_path,gen_list, data_in_tuples = True):
+
```
don't add extra enters. It makes very difficult to read the code.
and use 2 enters between function definitions. Aren't you using a linter?

rohangoel96 commented on test/tests.py:
```diff
@@ -1,29 +1,161 @@
+        pass
+
     def test_msg_number_graph(self):
+
```
don't add extra enters. It makes very difficult to read the code.
and use 2 enters between function definitions. Aren't you using a linter?

rohangoel96 commented on test/tests.py:
```diff
@@ -1,29 +1,161 @@
+
+    #since test are run alphabetically append a 'a' in test name to run it first
+    #since running this test later is creating error.
+    #needs to be fixed
```
what is this? please fix and then push

rohangoel96 commented on test/tests.py:
```diff
@@ -1,29 +1,161 @@
+        self.assertListEqual(resp_time, expected_resp_time, msg="resp_time not same as expected output")
+
+    def test_message_number_bins_csv(self):
+
```
don't add extra enters. It makes very difficult to read the code.
and use 2 enters between function definitions. Aren't you using a linter?

rohangoel96 commented on test/tests.py:
```diff
@@ -1,29 +1,161 @@
+        unpickle('node_out_degree2013-01-1-2013-08-31.pkl', expected_out_degree)
+        unpickle('node_total_degree2013-01-1-2013-08-31.pkl', expected_total_degree)
+
+        self.assertListEqual(out_degree, expected_out_degree, msg="Failed")
```
failed what? output should be test for outdegree for month of jan failed.. or something like that.

kaushik-rohit committed:
add end-to-end tests

Include end_to_end tests for functions inside the lib folder,
to test the generated output against expected output,
add data-driven-testing(ddt package) as requirement inside setup.py,
fix nicktracker to not return global references to function calls.
